### ISSUE TITLE
Unnecessary external AMD module required when aliased

### ISSUE THREAD

**Issue by stephenfhunt** [microsoft/TypeScript#1401]
Starting with 1.1, but not in 1.0.3, when a name from an import/required external module is aliased into the importing file's scope, the external module is always included in the define() dependencies of the JavaScript output.  This is incorrect in two cases:
- when the name is not actually used after the aliasing import
- when the name is only necessary at compile time for type-checking

The first can be resolved by removing the unused import.  The second is a bigger problem.  The import/require can not be removed because it is needed for type checking, but having a runtime dependency as well makes circular dependencies easy to introduce.  A solution is to avoid aliasing, but with long module paths that can be cumbersome.

Sample code:

Given a module looking something like:

```
export module elaborate.nested.mod.name {
    export class ReferredTo {
        doSomething() : void {
        }
    }
}
```

Which is imported by a module like:

```
import RT_ALIAS=require("ReferredTo");
import ReferredTo=RT_ALIAS.elaborate.nested.mod.name.ReferredTo;

export module ImportingModule {
    class UsesReferredType {
        constructor(private referred : ReferredTo) {}
    }
}
```

Using v1.0.3, and compiling with `-t ES5 -m AMD` the JS output begins with `define(["require", "exports"], function(require, exports) {`
whereas with v1.1 and later (this is with 1.3) the output begins `define(["require", "exports", "ReferredTo"], function (require, exports, RT_ALIAS) {`
and RT_ALIAS appears nowhere in the define() body, as expected.  

Likewise if the code is changed to

```
import RT_ALIAS=require("ReferredTo");

export module ImportingModule {
    class UsesReferredType {
        constructor(private referred : RT_ALIAS.elaborate.nested.mod.name.ReferredTo) {}
    }
}
```

with 1.3 the output begins `define(["require", "exports"], function (require, exports) {`

**Comment by stephenfhunt** [comment]
Could someone comment on the expected release for this fix?  Based on the milestone, I expected it to be in 1.4.1, but it wasn't.  I see the commit (59c73c5f34c58fa006194bf3bdbc60360308ff0b) in master, and when I build master things work properly, but the commit is not in the 1.4 tag or the release-1.4 branch.

**Comment by mhegazy** [comment]
This issue was marked as 1.4 by mistake. so it was never ported to that branch. the fix should be available in the next release (1.5), or off master if you do not mind garbing the latest.
